### ISSUE TITLE
tests failing

### ISSUE THREAD

**Issue by nsheff** [databio/GenomicDistributions#66]
devtools::test("~/code/GenomicDistributions")

are these bad tests, or finding code that needs to be fixed?

```
test_all.R:71: failure: featureDistribution
`nearestVec2` not equal to c(99, -901, -75, 124, -449).
3/5 mismatches (average diff: 1009)
[3] -1276 -  -75 == -1201
[4]   750 -  124 ==   626
[5] -1650 - -449 == -1201

test_all.R:124: error: Partitions
argument "query" is missing, with no default
Backtrace:
 1. GenomicDistributions::calcPartitions() tests/testthat/test_all.R:124:4
 2. GenomicDistributions::.validateInputs(...) R/partition-plots.R:142:4
 3. base::get(nms[i], envir = parent.frame(1)) R/utility.R:18:8
```

**Comment by joseverdezoto** [comment]
I think I know what the problem is. For some reason, 2 of the  values of the  `DTNearest()` output vector  aren't  the same as the ones calculated by hand.

**Comment by joseverdezoto** [comment]
https://github.com/databio/GenomicDistributions/blob/d4ef5541e17fa0eb906b5786589b98af66333a07/tests/testthat/test_all.R#L71

The vector I calculated by hand is actually `c(99, -901, -75, 750, -449)`
but the one produced by DTNearest is `c( 99, -901, -1276, 750, -1650)`

**Comment by nsheff** [comment]
so... which is correct ? :smile:

**Comment by joseverdezoto** [comment]
It's a bit confusing because when you run `DTNearest` as follows:
https://github.com/databio/GenomicDistributions/blob/d4ef5541e17fa0eb906b5786589b98af66333a07/tests/testthat/test_all.R#L62
the vector calculated by hand is the same as the function produced (no problem there).

The issue occurs when you do: 
https://github.com/databio/GenomicDistributions/blob/d4ef5541e17fa0eb906b5786589b98af66333a07/tests/testthat/test_all.R#L65
Looks like the closest midpoint to the third region midpoint of `coordDT2` is wrongly chosen.

**Comment by joseverdezoto** [comment]
Ok that makes sense. Also, the second error probably happened because that unit test was started but not completed. How do we know what functions should be tested in the `test_all.R` script?

**Comment by nsheff** [comment]
I'm not sure exactly what you mean. I guess the ideal is that every function is tested. We don't always hit that ideal, so it's just as many as possible.

Are you able to update these tests? I think once the tests are passing, I am planning to release this next version of GenomicDistributions. It's OK if we aren't comprehensive yet, but we should at least pass the tests we do have before release.

**Comment by joseverdezoto** [comment]
Yes, I updated the `DTNearest` unit test that was producing the error `test_all.R:71: failure: featureDistribution`. I also added another test since the second error `test_all.R:124: error: Partitions
argument "query" is missing, with no default` was produced because a unit test was just initialized but not completed. Those 2 tests are now passing. I believed I pushed those changes to the `dev_jose` branch last Friday and made a pull request.
